light/errors.go

```diff
@@ -20,6 +53,134 @@ func (e ErrOldHeaderExpired) Error() string {
 	return fmt.Sprintf("old header has expired at %v (now: %v)", e.At, e.Now)
 }
 
+type ErrTargetBlockHeightLessThanTrusted struct {
+	Target  int64
+	Trusted int64
+}
+
+func (e ErrTargetBlockHeightLessThanTrusted) Error() string {
+	return fmt.Sprintf("target block has a height lower than the trusted height (%d < %d)", e.Target, e.Trusted)
+}
+
+type ErrHeaderHeightNotMonotonic struct {
+	WantHeight int64
+	GetHeight  int64
+}
+
+func (e ErrHeaderHeightNotMonotonic) Error() string {
+	return fmt.Sprintf("expected new header height %d to be greater than one of old header %d", e.WantHeight, e.GetHeight)
+}
+
+type ErrHeaderTimeNotMonotonic struct {
+	WantTime time.Time
+	GetTime  time.Time
+}
+
+func (e ErrHeaderTimeNotMonotonic) Error() string {
+	return fmt.Sprintf("expected new header time %v to be after old header time %v", e.WantTime, e.GetTime)
```

Comment:
The argument order looks wrong here as well.

Reply:
resolved